### ISSUE TITLE
feature request: handle fits.gz files

### ISSUE THREAD

**Issue by teuben** [CARTAvis/carta#29]
Some archives deliver their data in fits.gz format, in the case I was using, they were able to get the sizes down to 50%.
Both fv and ds9 handle this transparently, and we're going to miss the boat of community acceptance if carta doesn't handle this transparently.
Yes, this means I had a case where I think carta does a better job than ds9, but I resorted to ds9 because I had too many files and didn't want to uncompress them.
In ds9 it's 8 steps (clicking/selecting) to get a spectrum on the screen.
In carta this is 1 step (arguably I would love to see that go to 0, if you are loading a cube, why give people an X and Y scan on the right, but not the more obvious Z scan, or give all 3)

**Comment by kswang1029** [comment]
You can make the default layout to “cube view” which has x, y and z scan. Or you can design your own layout and save it and make it as default via the preferences dialogue.

**Comment by kswang1029** [comment]
In addition, we have a plan to implement an option called “smart” layout which loads user-defined layouts based on image dimension/type.

**Comment by veggiesaurus** [comment]
> Some archives deliver their data in fits.gz format, in the case I was using, they were able to get the sizes down to 50%.
> Both fv and ds9 handle this transparently, and we're going to miss the boat of community acceptance if carta doesn't handle this transparently.

I agree that we should support this. Unless I'm mistaken, compressed FITS files are handled correctly by cfitsio. So we might just need to investigate what's stopping casacore from supporting this. @teuben can you send us an example file? I remember trying _some_ fits.gz files that worked properly in CARTA, and others that didn't.

**Comment by teuben** [comment]
ok, here's a good challenge.   I have two files in my https://www.astro.umd.edu/~teuben/data/
The cube is hidden as an image in the first extension, the m0 file is a true classic fits file.
Salient detail:    ds9 will not correctly overlay (match WCS) these two, I will need to contact the ds9 author about this.
These data are from Hershel, they have a pretty complex hierarchy of fits.gz files, where some fits files are tables referring to the datafits files. nutty.
